checkov/common/graph/checks_infra/registry.py

```diff
@@ -1,20 +1,32 @@
+from typing import List, Dict, Any
+
+from networkx import DiGraph
+
+from checkov.common.graph.checks_infra.base_check import BaseGraphCheck
+from checkov.common.graph.checks_infra.base_parser import BaseGraphCheckParser
 from checkov.common.models.enums import CheckResult
 from checkov.runner_filter import RunnerFilter
 
 
 class BaseRegistry:
-    def __init__(self, parser):
-        self.checks = []
+    def __init__(self, parser: BaseGraphCheckParser) -> None:
+        self.checks: List[BaseGraphCheck] = []
         self.parser = parser
 
     def load_checks(self):
         raise NotImplementedError
 
-    def run_checks(self, graph_connector, runner_filter: RunnerFilter):
+    def run_checks(
+        self, graph_connector: DiGraph, runner_filter: RunnerFilter
+    ) -> Dict[BaseGraphCheck, List[Dict[str, Any]]]:
         check_results = {}
         for check in self.checks:
-            if runner_filter.checks and check.id not in runner_filter.checks \
-                    or runner_filter.skip_checks and not runner_filter.should_run_check(check.id):
+            if (
+                runner_filter.checks
+                and check.id not in runner_filter.checks
+                or runner_filter.skip_checks
+                and not runner_filter.should_run_check(check.id)
```

Comment:
@gruebel `and` and `or` inline is not a good idea... Mind adding parenthesis to make the order more explicit?